share-on-mastodon.js

```diff
@@ -197,120 +208,143 @@ export class shareOnMastodon extends HTMLElement {
+      this.shadowRoot.querySelector("#instance-jargon").innerHTML =
+        this.getAttribute("data-jargon-text");
+    }
 
     // Get the default url to use as placeholder text from attribute if available
-    if (this.getAttribute('default_url')) {
-      this.shadowRoot.querySelector('#instanceInput').placeholder = this.getAttribute('default_url');
-    };
+    if (this.getAttribute("data-default-url")) {
+      this.shadowRoot.querySelector("#instance-input").placeholder =
+        this.getAttribute("data-default-url");
+    }
 
     // Get the share button text from attribute if available
-    if (this.getAttribute('share_text')) {
-      this.shadowRoot.querySelector('#shareText').innerHTML = this.getAttribute('share_text');
-    };
+    if (this.getAttribute("data-share-text")) {
+      this.shadowRoot.querySelector("#share-text").innerHTML =
+        this.getAttribute("data-share-text");
+    }
   }
 
   openModal() {
-  // Prefill the form with the user's previously-specified Mastodon instance, if applicable
-  if (localStorage['mastodon_instance']) {
-    this.shadowRoot.querySelector('#instanceInput').value = localStorage['mastodon_instance'];
-  }
+    // Prefill the form with the user's previously-specified Mastodon instance, if applicable
+    const storedInstance = localStorage.getItem("mastodonInstance");
+    if (storedInstance) {
+      this.shadowRoot.querySelector("#instance-input").value = storedInstance;
+    }
 
-  // open Dialog as modal
-  this.shadowRoot.querySelector('#mastodonModal').showModal();
+    // open Dialog as modal
+    this.shadowRoot.querySelector("#mastodon-modal").showModal();
   }
 
-  validate_before_share() {
-    if (this.shadowRoot.querySelector('#instanceInput').checkValidity()) {
-      this.share_on_mastodon();
+  urlCheck() {
+    let val = this.shadowRoot.querySelector("#instance-input").value;
+    if (!val == "" && !val.startsWith("https://")) {
+      this.shadowRoot.querySelector("#instance-input").value = "https://" + val;
     }
+    this.shadowRoot.querySelector("#instance-input").checkValidity();
   }
 
-  share_on_mastodon() {
+  shareOnMastodon() {
     // Set instance from the modal text input
-    let instance = this.shadowRoot.querySelector('#instanceInput').value;
+    let instance = this.shadowRoot.querySelector("#instance-input").value;
 
     // check, format, and cache instance
     if (instance) {
       // Handle URL formats
-      if ( !instance.startsWith("https://") && !instance.startsWith("http://") )
+      if (!instance.startsWith("https://") && !instance.startsWith("http://"))
         instance = "https://" + instance;
 
       // Handle ending slash
-      if ( !instance.endsWith("/") )
-        instance = instance + "/";
+      if (!instance.endsWith("/")) instance = instance + "/";
 
       // Cache the instance/domain for future requests
-      localStorage['mastodon_instance'] = instance;
-    };
+      localStorage.setItem("mastodonInstance", instance);
```

Comment:
```suggestion
      localStorage?.setItem("mastodonInstance", instance);
```
it's rare but you can technically disable client-side storage. Best to guard against that!